src/app/clusters/level-control/level-control.cpp

```diff
@@ -225,11 +225,11 @@ class DefaultLevelControlSceneHandler : public scenes::DefaultSceneHandlerImpl
             switch (decodePair.attributeID)
             {
             case Attributes::CurrentLevel::Id:
-                level = static_cast<uint8_t>(decodePair.attributeValue);
+                level = decodePair.valueUnsigned8.Value();
```

Comment:
What ensures that this does not crash (i.e. that decodePair.valueUnsigned8 has a value)?

Reply:
As above, added Added a check on `valueUnsigned8.HasValue()` before, preventing storage of invalid attributes to be implemented in the future.